core/content/content.go

```diff
@@ -163,6 +163,9 @@ type Writer interface {
 
 	// Truncate updates the size of the target blob
 	Truncate(size int64) error
+
+	// Sync flushes the in-flight writes to the disk (when applicable)
+	Sync() error
```

Comment:
This interface should not change, `Sync() error` can either be added to a new interface here or defined next to where it needs to be called. For new and optional functions, this should just use a type assert and called if available. In this case only the filesystem writer would need a Sync function.

Reply:
Eventually, we need to call `namespacedWriter.w.fp.Sync()` from the `namedspacedWriter.Commit()` to actually flush the writes.  But these are all private members of the class hiding behind the interface, so I don't see a clean way to do it without modifying the interface.  Can you elaborate how to expose `namespacedWriter.w.fp.Sync()` just for the filesystem writer?

Reply:
Oh, do you suggest to check the actual type (behind the interface) dynamically and then call the Sync function when it is a filesystem writer?

Reply:
> Oh, do you suggest to check the actual type (behind the interface) dynamically and then call the Sync function when it is a filesystem writer?

Yes

Somewhere during namespacedWriter Commit before the transaction you could have something like
```
if syncer, ok := nw.w.(Syncer); ok {
  syncer.Sync()
}
```

Reply:
Done.  I will test the change.

Reply:
Tested and it works.  Thanks for reviewing the PR.